src/course-checklist/ChecklistSection/ChecklistItemBody.jsx

```diff
@@ -1,65 +1,89 @@
-import React from 'react';
 import PropTypes from 'prop-types';
+import { useNavigate } from 'react-router';
 import { injectIntl, intlShape, FormattedMessage } from '@edx/frontend-platform/i18n';
 import {
   ActionRow,
   Button,
   Hyperlink,
   Icon,
 } from '@openedx/paragon';
+import { useSelector } from 'react-redux';
 import { CheckCircle, RadioButtonUnchecked } from '@openedx/paragon/icons';
+
+import { getWaffleFlags } from '../../data/selectors';
 import messages from './messages';
 
 const ChecklistItemBody = ({
+  courseId,
   checkId,
   isCompleted,
   updateLink,
   // injected
   intl,
-}) => (
-  <ActionRow>
-    <div className="mr-3" id={`icon-${checkId}`} data-testid={`icon-${checkId}`}>
-      {isCompleted ? (
-        <Icon
-          data-testid="completed-icon"
-          src={CheckCircle}
-          className="text-success"
-          style={{ height: '32px', width: '32px' }}
-          screenReaderText={intl.formatMessage(messages.completedItemLabel)}
-        />
-      ) : (
-        <Icon
-          data-testid="uncompleted-icon"
-          src={RadioButtonUnchecked}
-          style={{ height: '32px', width: '32px' }}
-          screenReaderText={intl.formatMessage(messages.uncompletedItemLabel)}
-        />
-      )}
-    </div>
-    <div>
-      <div>
-        <FormattedMessage {...messages[`${checkId}ShortDescription`]} />
+}) => {
+  const waffleFlags = useSelector(getWaffleFlags);
+  const navigate = useNavigate();
+
+  const handleClick = (e, url) => {
+    e.preventDefault();
+
+    if (waffleFlags?.useNewUpdatesPage) {
+      navigate(`/course/${courseId}/course_info`);
+    } else {
+      window.location.href = url;
```

Comment:
I don't like that these "checklist" URLs are mostly defined here:

https://github.com/openedx/frontend-app-authoring/blob/cff1177ae93db0f63377ef387fca232b24e47c0b/src/course-checklist/utils.js#L3-L9

but depending on the waffle flag, they may be defined here in `ChecklistItemBody.jsx`. Also, you seem to be just linking every course update to `course_info` rather than the more specific link. Please refactor to define ALL the URLs in one place, something like:

```ts
const getUpdateLinks = (courseId: string, useLegacyUrl = false) => ({
  welcomeMessage: useLegacyUrl ? `${getConfig().STUDIO_BASE_URL}/course_info/${courseId}` : `/course_info/${courseId}`,
```

Also, please just use react-router's `<Link>` component instead of implementing a `handleClick` function and using `navigate()`

Reply:
Implemented in follow-up PR: [`885ecab` (#1372)](https://github.com/openedx/frontend-app-authoring/pull/1372/commits/885ecaba6852f91719b84a62ea9b660c423b25ee)
Thanks!